libcaf_core/caf/flow/observable_decl.hpp

```diff
@@ -173,6 +173,15 @@ class observable {
     return reduce(T{}, [](T x, T y) { return x + y; });
   }
 
+  /// Adds the output of fn() to the beginning of current observable.
+  template <class Input>
+  auto start_with(Input value) {
+    if constexpr (is_observable_v<Input>)
+      return std::move(value).merge(*this);
```

Comment:
Please read up on the semantics of the operators. Merge will not make sure that the result will start with the inputs. It'll *merge* the things, not concatenate them. The tests only pass because the data sets are so small that the merge operator just forwards everything in one shot.

```c++
      return std::move(value).concat(*this);
```